src/components/Form/index.tsx

```diff
@@ -0,0 +1,86 @@
+import RjsForm from '@rjsf/mui';
+import ajvValidator from '@rjsf/validator-ajv8';
+import { FormProps as RjsFormProps } from '@rjsf/core';
+import { useCallback } from 'react';
+import { Box, Button, ButtonProps } from '@mui/material';
+import { PasswordWidget } from './PasswordWidget';
+import { TextWidget } from './TextWidget';
+import { TextareaWidget } from './TextareaWidget';
+import { SelectWidget } from './SelectWidget';
+import { DateTimeWidget } from './DateTimeWidget';
+import { UpDownWidget } from './UpDownWidget';
+
+const internalWidgets: {
+	[k: string]: any;
+} = {
+	PasswordWidget,
+	EmailWidget: TextWidget,
+	TextWidget,
+	TextareaWidget,
+	SelectWidget,
+	UpDownWidget,
+	DateTimeWidget,
+};
+
+export interface MuiFormProps extends Omit<RjsFormProps, 'validator'> {
+	/** If true, do not display the form submit button */
+	hideSubmitButton?: boolean;
+	/** Properties that are passed to the submit button, these are the same props used for the [`Button`](#button) component */
+	submitButtonProps?: ButtonProps;
+	/** If passed, it will show a secondary button, these are the same props used for the [`Button`](#button) component */
+	secondaryButtonProps?: ButtonProps;
+
+	validator?: RjsFormProps['validator'];
+}
+
+export const MuiForm: React.FC<React.PropsWithChildren<MuiFormProps>> = ({
+	hideSubmitButton,
+	submitButtonProps,
+	secondaryButtonProps,
+	validator = ajvValidator,
+	widgets,
+	fields,
+	children,
+	...otherProps
+}) => {
+	const handleKeyUpDown = useCallback(
+		(e: React.KeyboardEvent<HTMLDivElement>) => {
+			if (e.key === 'Enter' && !hideSubmitButton) {
+				e.stopPropagation();
+			}
+		},
+		[hideSubmitButton],
+	);
+	return (
+		<Box onKeyDown={handleKeyUpDown} onKeyUp={handleKeyUpDown}>
+			<RjsForm
+				validator={validator}
+				showErrorList={false}
+				onChange={(data) => {
+					otherProps.onChange?.(data);
+				}}
+				widgets={{ ...internalWidgets, ...(widgets || {}) }}
+				{...otherProps}
+			>
+				{hideSubmitButton && <span />}
+
+				{!hideSubmitButton && (
+					<Button
+						{...submitButtonProps}
+						color="customBlue"
+						variant="contained"
+						type="submit"
+					/>
+				)}
+				{!!secondaryButtonProps && (
+					<Button
+						{...secondaryButtonProps}
+						variant="outlined"
+						color="customBlue"
+					/>
+				)}
```

Comment:
These two Buttons need to be reversed so that the primary (submit) button is on the right (which lines up with our Dialog UX)

Reply:
Also I think the buttons should align right
Not sure what standard/ideal UX is on this point though